fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java

```diff
@@ -279,4 +281,38 @@ private void appendChainedParm(QueryParameter param, StringBuilder returnString)
         }
     }
 
+    /*
+     * creates a reverse chained parameter
+     *
+     * @param param
+     *
+     * @param returnString
+     */
+    private void appendReverseChainedParm(QueryParameter param, StringBuilder returnString) {
+        // Build initial string: "_has:<reference-resource-type>:<refernece-search-parm>:
+        returnString.append(SearchConstants.HAS)
+                    .append(SearchConstants.COLON_DELIMITER)
+                    .append(param.getModifierResourceTypeName())
+                    .append(SearchConstants.COLON_DELIMITER)
+                    .append(param.getCode())
+                    .append(SearchConstants.COLON_DELIMITER);
+
+        // Loop through chained parms to build chained strings
+        for (QueryParameter revChainParam : param.getChain()) {
+            if (revChainParam.isReverseChained()) {
+                returnString.append(SearchConstants.HAS)
+                            .append(SearchConstants.COLON_DELIMITER)
+                            .append(revChainParam.getModifierResourceTypeName())
+                            .append(SearchConstants.COLON_DELIMITER)
+                            .append(revChainParam.getCode())
+                            .append(SearchConstants.COLON_DELIMITER);
+            } else if (revChainParam.isChained()) {
+                appendChainedParm(revChainParam, returnString);
+                returnString.append(SearchConstants.CHAINED_PARAMETER_CHARACTER);
+            } else {
+                appendNormalParameter(revChainParam, returnString);
```

Comment:
If I follow... 

1st part
_has:Patient:patient:

It's always going to run the param.getChain (it'll have at least one entry)
Then loops... 

OK - this makes sense to me - I just needed to write this out, leaving here for others as they review.